shared/pkg/scanner/trivy/scanner.go

```diff
@@ -198,6 +199,32 @@ func (a *Scanner) Run(sourceType utils.SourceType, userInput string) error {
 			return
 		}
 
+		switch sourceType {
+		case utils.DOCKERARCHIVE:
+			trivyOptions.ImageOptions.Input = userInput
+		case utils.OCIARCHIVE:
+			tmpDir, err := os.MkdirTemp("", "")
+			if err != nil {
+				a.setError(fmt.Errorf("unable to create temp directory: %w", err))
+				return
+			}
+			defer func() {
+				err := os.RemoveAll(tmpDir)
+				if err != nil {
+					a.logger.WithError(err).Errorf("unable to remove temp directory %s", tmpDir)
+				}
+			}()
+			err = utilsTrivy.UntarToDirectory(userInput, tmpDir)
```

Comment:
can you explain why the untar is needed here and not in the docker-archive one?

Reply:
Trivy doesn't support OCI archive for some reason, but it does support OCI directories so we have to untar it first in order to use it. I'll add support for oci-dir too.